### ISSUE TITLE
Mesos redirects to a wrong port of the leader

### ISSUE THREAD

**Issue by OleksandrBerchenko** [CiscoCloud/microservices-infrastructure#148]
When I load Mesos UI (running on port 5050) of a host, which is not a leader, it redirects me to the leader, but to a wrong port: 15050.  This should be fixed.

```
[oberchen@gluon-int-edge-01 microservices-infrastructure]$ grep -ri 15050 *
roles/mesos/files/mesos-leader-iptables.tmpl:iptables -D INPUT -p tcp --dport 15050 -j MESOSLEADER
roles/mesos/files/mesos-leader-iptables.tmpl:iptables -A MESOSLEADER -p tcp --dport 15050 -s {{ .Address }} -j ACCEPT
roles/mesos/files/mesos-leader-iptables.tmpl:iptables -A MESOSLEADER -p tcp --dport 15050 -s {{ .Address }} -j ACCEPT
roles/mesos/files/mesos-leader-iptables.tmpl:iptables -A MESOSLEADER -p tcp --dport 15050 -d 127.0.0.1 -j ACCEPT
roles/mesos/files/mesos-leader-iptables.tmpl:iptables -A MESOSLEADER -p tcp --dport 15050 -j DROP
roles/mesos/files/mesos-leader-iptables.tmpl:iptables -A INPUT -p tcp --dport 15050 -j MESOSLEADER
roles/mesos/files/mesos-leader.nginx:                   proxy_pass http://localhost:15050/;
roles/mesos/defaults/main.yml:mesos_leader_port: 15050
Vagrantfile:  config.vm.network :forwarded_port, guest: 15050, host: 15050  # Mesos leader UI
```

```
[oberchen@gluon-int-edge-01 microservices-infrastructure]$ grep -ri 5050 * | grep -v 15050
docs/getting_started/index.rst:*  `mesos-leader`_ on port 5050
docs/getting_started/index.rst:.. _mesos-leader: http://host-01:5050
docs/getting_started/openstack.rst:   Mesos            TCP      5050/5051
docs/getting_started/openstack.rst:   nova secgroup-add-rule default tcp 5050 5050 0.0.0.0/0
roles/mesos/files/mesos-leader.nginx:           listen {{ env "NGINX_PUBLIC_IP" }}:5050 ssl;
roles/mesos/README.rst:   default: ``5050``
Vagrantfile:  config.vm.network :forwarded_port, guest: 5050, host: 5050  # Mesos leader
```

Thanks!

**Comment by ChrisAubuchon** [comment]
In order to implement authentication for the Mesos UI it is necessary to have it actually listen on a different port and put a proxy in front. Unfortunately the redirection is internal to Mesos. There is a Mesos patch in the works that will help with this issue. https://issues.apache.org/jira/browse/MESOS-809. Until then our choices are:
1. Remove authentication from Mesos.
2. Live with the incorrect redirection until Mesos is patched.

**Comment by OleksandrBerchenko** [comment]
Thanks for the link.  I believe the right choice is:

2. Live with the incorrect redirection until Mesos is patched.

Let's have this issue opened.

Thanks!

**Comment by siddharthist** [comment]
Fixed in #708
